routers/api/v1/admin/user.go

```diff
@@ -117,11 +116,8 @@ func CreateUser(ctx *context.APIContext) {
 	}
 
 	overwriteDefault := &user_model.CreateUserOverwriteOptions{
-		IsActive: util.OptionalBoolTrue,
-	}
-
-	if form.Restricted != nil {
-		overwriteDefault.IsRestricted = util.OptionalBoolOf(*form.Restricted)
+		IsActive:     optional.Some(true),
```

Comment:
I couldn't find this line has better readability than before.

Reply:
Maybe it could be rename to `optional.Value`

```
IsActive:     optional.Value(true),
```

Reply:
Or maybe `Nullable`, a `Nullable` is a generic type which can combine a Go basic type and nil.

Reply:
See here for more background: https://github.com/go-gitea/gitea/pull/28733#discussion_r1445272944

Reply:
https://github.com/go-gitea/gitea/pull/29341

Reply:
> See here for more background: [#28733 (comment)](https://github.com/go-gitea/gitea/pull/28733#discussion_r1445272944)

I can understand that many languages use "Some", but most languages using "Some" because they have "match" pattern like :

```
value = option.Match(
  some: ...
```

For languages like C++, which don't have "match" pattern , they never use "some", they only use "value": https://en.cppreference.com/w/cpp/utility/optional/optional


To me, either is fine. I am pretty neutral.

Reply:
I'm used to Some and None from rust ...
... in the ned it's just strings so golang does not care how we name it

Reply:
Anyway if this needs discussion, we should create a dedicated issue for that. In genneral the optional package itselv works and do cover NoneBool just fine